app/models/work.rb

```diff
@@ -474,13 +476,18 @@ def set_challenge_info
     end
   end
 
+  # If this is fulfilling a challenge claim, add the collection.
+  #
+  # Unlike set_challenge_info, we don't automatically add the prompter as a
+  # recipient, because (a) some prompters are anonymous, so there has to be a
+  # prompter notification (separate from the recipient notification) ensuring
+  # that anonymous prompters are notified, and (b) if the prompter is not
+  # anonymous, they'll receive two notifications with roughly the same info
+  # (gift notification + prompter notification).
```

Comment:
This method was never called before, so having it now would likely fix [AO3-3556](https://otwarchive.atlassian.net/browse/AO3-3556) as well, we'll see.

----

Notes for me: @sarken said we handle prompter notification at:

https://github.com/otwcode/otwarchive/blob/bd4f323003a7f63d0e708ff334fd666e4760601a/lib/creatable.rb#L100-L109

and also (for unrevealed works specifically):

https://github.com/otwcode/otwarchive/blob/59544920efb52c5ffedb6fb062c7d2388264feac/app/models/collection_item.rb#L310-L313